src/components/Web3Status/index.tsx

```diff
@@ -147,13 +148,36 @@ function Web3StatusInner() {
   const { hasPendingActivity, pendingActivityCount } = usePendingActivity()
 
   const displayName = useMemo(() => (account ? ENSName || shortenAddress(account) : null), [account, ENSName])
-  const lastDisplayName = useRef(localStorage.getItem(selectedWalletDisplayKey) ?? null)
+  const meta = useRef(getConnectionMeta())
   useEffect(() => {
     if (displayName) {
-      // It takes a moment to load the ENSName, so set the lastDisplayName (for next time) to the account to reduce layout shift.
-      localStorage.setItem(selectedWalletDisplayKey, shortenAddress(account))
+      const meta: ConnectionMeta = {
+        type: connection.type,
+        display: displayName,
+        displayType: ENSName ? ConnectionDisplayType.ENSName : ConnectionDisplayType.Address,
+      }
+      setConnectionMeta(meta)
     }
-  }, [account, displayName])
+  }, [ENSName, account, connection.type, displayName])
+  const isDisplayNameLoading = Boolean(
+    account && meta.current?.displayType === ConnectionDisplayType.ENSName && ENSLoading
```

Comment:
Is there a situation where we would be loading an ENSName without an `account` or `meta.current?.displayType !== ConnectionDisplayType.ENSName`?

Reply:
not that i can think of, no